src/dos/cdrom.h

```diff
@@ -100,11 +100,11 @@ class CDROM_Interface
 	virtual bool GetAudioSub        (unsigned char& attr, unsigned char& track, unsigned char& index, TMSF& relPos, TMSF& absPos) = 0;
 	virtual bool GetAudioStatus     (bool& playing, bool& pause) = 0;
 	virtual bool GetMediaTrayStatus (bool& mediaPresent, bool& mediaChanged, bool& trayOpen) = 0;
-	virtual bool PlayAudioSector    (uint64_t start, uint64_t len) = 0;
+	virtual bool PlayAudioSector    (const uint32_t start, uint32_t len) = 0;
```

Comment:
Let's look at it relative to the function body: `const` applied to **any** function argument has consistent behavior, and documents the fact that the inbound variable shall have it's value locked read-only for the entire duration of the function.  New maintainers can immediately trust that fact, even big nasty winding functions.  

So because it's behavior is consistent (even for scalars), it has value right there - it  documents the use of the variable through the life-time of the function.

When I come back to my own code (often after some time and I'm practically new to it again), I immediately know that my non-const arguments are being changed in the code, perhaps as a decrementing counter or for some dual-purpose.

In my mind, this is all separate from whatever side-effects the function can impart outside it's scope via pointer arguments and references.  

In perfect code, you've got read-only variables inboud to the function, and the new result(s) outbound from the function. The language under the hood should take care of efficiently minimizing copies-of-new-variables and so on, so it behaves.   

``` c++
#include <iostream>

using namespace std;

int change_test(const int val) {
    val++;
    return val;
}

int main()
{
    cout << change_test(1);
    return 0;
}
```

Returns:

```
main.cpp: In function ‘int test(int)’:
main.cpp:14:9: error: assignment of read-only parameter ‘d’
```

Reply:
Overall I am a big fan of using const whenever possible - I just find it a bit weird when used for non-pointer, non-reference function arguments in C/C++ - so it's very specific user complaint ;)

It was only a style suggestion, so we can close this topic and leave the code as it is now :)